rust-src/id/src/identity_provider.rs

```diff
@@ -246,13 +246,462 @@ pub fn validate_request<P: Pairing, C: Curve<Scalar = P::ScalarField>>(
+//         commitment_2: *pre_id_obj
+//             .cmm_prf_sharing_coeff
+//             .first()
+//             .expect("Precondition checked."),
+//         cmm_key_1:    commitment_key_prf,
+//         cmm_key_2:    *ar_ck,
+//     };
+//     let witness_prf_same = pre_id_obj.poks.commitments_prf_same;
+
+//     let h_in_exponent =
+// *context.global_context.encryption_in_exponent_generator();
+//     let prf_verification = compute_prf_sharing_verifier(
+//         ar_ck,
+//         &pre_id_obj.cmm_prf_sharing_coeff,
+//         &pre_id_obj.ip_ar_data,
+//         &context.ars_infos,
+//         &h_in_exponent,
+//     );
+//     let (prf_sharing_verifier, prf_sharing_witness) = match prf_verification
+// {         Some(v) => v,
+//         None => return Err(Reason::WrongArParameters),
+//     };
+
+//     let verifier = AndAdapter {
+//         first:  id_cred_sec_verifier,
+//         second: id_cred_sec_eq_verifier,
+//     };
+//     let verifier = verifier
+//         .add_prover(verifier_prf_same)
+//         .add_prover(prf_sharing_verifier);
+//     let witness = AndWitness {
+//             w1: AndWitness {
+//                 w1: AndWitness {
+//                     w1: id_cred_sec_witness,
+//                     w2: id_cred_sec_eq_witness,
+//                 },
+//                 w2: witness_prf_same,
+//             },
+//             w2: prf_sharing_witness
+//     };
+//     let proof = SigmaProof {
+//         challenge: pre_id_obj.poks.challenge,
+//         witness,
+//     };
+//     let bulletproofs = &pre_id_obj.poks.bulletproofs;
+//     for ((ar_identity, ar_data), proof) in pre_id_obj
+//         .ip_ar_data
+//         .iter()
+//         // .zip(context.ars_infos.values())
+//         .zip(bulletproofs.iter())
+//     {
+//         let ciphers = ar_data.enc_prf_key_share;
+//         let ar_info = match context.ars_infos.get(ar_identity) {
+//             Some(x) => x,
+//             None => return Err(Reason::IncorrectProof),
+//         };
+//         let pk: C = ar_info.ar_public_key.key;
+//         let keys: CommitmentKey<C> = CommitmentKey {
+//             g: h_in_exponent,
+//             h: pk,
+//         };
+//         let gens = &context.global_context.bulletproof_generators().take(32 *
+// 8);         let commitments = ciphers.iter().map(|x|
+// Commitment(x.1)).collect::<Vec<_>>();         transcript.append_message(b"
+// encrypted_share", &ciphers);         if verify_efficient(&mut transcript, 32,
+// &commitments, &proof, gens, &keys).is_err() {             return
+// Err(Reason::IncorrectProof);         }
+//     }
+
+//     transcript.append_message(b"bulletproofs", &bulletproofs);
+//     if verify(&mut transcript, &verifier, &proof) {
+//         Ok(())
+//     } else {
+//         Err(Reason::IncorrectProof)
+//     }
+// }
+
 /// Sign the given pre-identity-object to produce an identity object.
-pub fn sign_identity_object<
+pub fn sign_identity_object_old<
```

Comment:
Will this be removed? Otherwise please rename to `_v0`.

Reply:
To be removed.